src/containers/MemoryGame/index.jsx

```diff
@@ -1,12 +1,44 @@
-import React from "react";
+import React, { useState } from "react";
 import "./index.css";
-import MemoryGameCards from "../../components/memorygame/MemoryGameCards/index";
+import { Col, Row } from "antd";
+import { useTranslation } from "react-i18next";
+import Confetti from "react-dom-confetti";
+import confettiConfig from "../../components/memorygame/confettiConfig";
+import MemoryGameFactList from "../../components/memorygame/MemoryGameFactList";
+import MemoryGameBoard from "../../components/memorygame/MemoryGameBoard";
 
 const MemoryGame = () => {
+  const [t] = useTranslation();
+  const [isGameCompleted, setIsGameCompleted] = useState(false);
+  // finishedCards is being used to display environmental facts.
+  // It is an array of object(states).Each object is a successfully matched card state.
+  // When a match is made successfully only state of first card gets pushed inside finishedCards.
+  // Because matched cards have same state. Otherwise same fact would be displayed twice.
+  const [finishedCards, setFinishedCards] = useState([]);
```

Comment:
you dont need to change this here (please don't in any of the PRs you sent out because i don't want to rereview the PRs, do it in a separate PR if you want) but unless you need the whole state, the cleanest way to implement this would not be to push around card states, but probably to only store the imageKey of finished cards. then using the image key you would look up the facts related to that image

basically you can think that in your current structure of passing around data you are duplicating/denormalizing data a lot, which is risky esp. in a language like javascript where you are mutating some parts of the state etc.

in general a warning sign would be any code that says "i duplicated these items but i only need to keep one of them, but you know, all of them are the same." (except that a HUGE class of bugs is going to be when the duplicates actually differ in some way you didn't expect). storing only the image key and only looking up the facts by the image key guarantees that this is not possible.

finished cards -> store image key -> look up facts by image key for finished fact list

Reply:
I noted this. After finishing my issues I will try to help others. If we still have time after that I will implement this.